### ISSUE TITLE
sdbench fails on assertion

### ISSUE THREAD

**Issue by akamaus** [cmu-db/peloton#571]
I'm having troubles running sdbench.
```
./bin/sdbench                                                                                                                                                                
2017-03-15 17:57:45 [src/main/sdbench/sdbench_configuration.cpp:102:ValidateIndexUsageType] INFO  - index_usage_type  : PARTIAL_FAST
2017-03-15 17:57:45 [src/main/sdbench/sdbench_configuration.cpp:230:ValidateWriteRatio] INFO  - write_ratio : READ_ONLY
2017-03-15 17:57:45 [src/main/sdbench/sdbench_configuration.cpp:130:ValidateQueryComplexityType] INFO  - query_complexity_type  : SIMPLE
2017-03-15 17:57:45 [src/main/sdbench/sdbench_configuration.cpp:152:ValidateWriteComplexityType] INFO  - write_complexity_type : SIMPLE
2017-03-15 17:57:45 [src/main/sdbench/sdbench_configuration.cpp:172:ValidateScaleFactor] INFO  - scale_factor : 100
2017-03-15 17:57:45 [src/main/sdbench/sdbench_configuration.cpp:220:ValidateAttributeCount] INFO  - column_count : 200
2017-03-15 17:57:45 [src/main/sdbench/sdbench_configuration.cpp:266:ValidateTuplesPerTileGroup] INFO  - tuples_per_tilegroup : 1000
2017-03-15 17:57:45 [src/main/sdbench/sdbench_configuration.cpp:248:ValidateTotalOps] INFO  - total_ops : 10
2017-03-15 17:57:45 [src/main/sdbench/sdbench_configuration.cpp:257:ValidatePhaseLength] INFO  - phase_length : 10
2017-03-15 17:57:45 [src/main/sdbench/sdbench_configuration.cpp:211:ValidateSelectivity] INFO  - selectivity : 0.001
2017-03-15 17:57:45 [src/main/sdbench/sdbench_configuration.cpp:202:ValidateProjectivity] INFO  - projectivity : 0.010
2017-03-15 17:57:45 [src/main/sdbench/sdbench_configuration.cpp:182:ValidateLayout] INFO  - layout  : ROW
2017-03-15 17:57:45 [src/main/sdbench/sdbench_configuration.cpp:344:ValidateIndexCountThreshold] INFO  - index_count_threshold : 10
2017-03-15 17:57:45 [src/main/sdbench/sdbench_configuration.cpp:355:ValidateIndexUtilityThreshold] INFO  - index_utility_threshold : 0.25
2017-03-15 17:57:45 [src/main/sdbench/sdbench_configuration.cpp:365:ValidateWriteRatioThreshold] INFO  - write_ratio_threshold : 0.75
2017-03-15 17:57:45 [src/main/sdbench/sdbench_configuration.cpp:285:ValidateDurationOfPause] INFO  - duration_of_pause : 100
2017-03-15 17:57:45 [src/main/sdbench/sdbench_configuration.cpp:276:ValidateDurationBetweenPauses] INFO  - duration_between_pauses : 10000
2017-03-15 17:57:45 [src/main/sdbench/sdbench_configuration.cpp:296:ValidateAnalyzeSampleCountThreshold] INFO  - analyze_sample_count_threshold : 100
2017-03-15 17:57:45 [src/main/sdbench/sdbench_configuration.cpp:307:ValidateMaxTileGroupsIndexed] INFO  - max_tile_groups_indexed : 10
2017-03-15 17:57:45 [src/main/sdbench/sdbench_configuration.cpp:324:ValidateQueryConvergenceThreshold] INFO  - convergence_query_threshold : 200
2017-03-15 17:57:45 [src/main/sdbench/sdbench_configuration.cpp:334:ValidateVariabilityThreshold] INFO  - variability_threshold : 100
2017-03-15 17:57:45 [src/main/sdbench/sdbench_configuration.cpp:369:ValidateMultiStage] INFO  - multi_stage: 0
2017-03-15 17:57:45 [src/main/sdbench/sdbench_configuration.cpp:373:ValidateHolisticIndexing] INFO  - holistic_indexing : 0
sdbench: /home/vyal/src/peloton/src/concurrency/timestamp_ordering_transaction_manager.cpp:874: virtual peloton::ResultType peloton::concurrency::TimestampOrderingTransactionManager::CommitTransaction(peloton::concurrency::Transaction*): Assertion `(tile_group_header->GetTransactionId(tuple_slot) == current_txn->GetTransactionId())' failed.
```
Also I noticed `current_txn->txn_id_` equals 4294967296 == 2**32 which is strange given it's uint64_t type.

**Comment by jarulraj** [comment]
I believe that this bug is related to CC changes. @yingjunwu your thoughts? Should we change the CC from Timestamp Ordering?

**Comment by yingjunwu** [comment]
@jarulraj I've checked the code, and there's nothing wrong with the CC. It is also reasonable to get a transaction id equal to 2**32, which is essentially 1<<32 | 0x0 (the higher 32 bit is epoch id, and the lower 32 bit is unique id). The assertion fails because sdbench uses pretty old APIs. More specifically, it even did not invoke any transaction API during table reloading. I have fixed this problem in PR #574.

However, I want to point out that there are other bugs in this benchmark. When running the benchmark, I encountered the following assertion failure:

```
sdbench: /home/yingjun/peloton/src/storage/data_table.cpp:808: void peloton::storage::DataTable::DropIndexWithOid(const oid_t&): Assertion `(index_offset < indexes_.GetSize())' failed.
Aborted (core dumped)
```

It seems that the index tuner never adds new indexes to the data table, and hence `indexes_.GetSize()` is always equal to 1. As it is related to index tuner, I think you could fix this bug.
